lms/djangoapps/grades/api/v1/views.py

```diff
@@ -0,0 +1,192 @@
+    **GET Parameters**
+        A GET request may include the following parameters.
+        * course_id: (required) A string representation of a Course ID.
+    **GET Response Values**
+        If the request for information about the course grade
+        is successful, an HTTP 200 "OK" response is returned.
+        The HTTP 200 response has the following values.
+        * username: A string representation of a user's username passed in the request.
+        * course_id: A string representation of a Course ID.
+        * passed: Boolean representing whether the course has been
+                  passed according the course's grading policy.
+        * percent: A float representing the overall grade for the course
+        * letter_grade: A letter grade as defined in grading_policy (e.g. 'A' 'B' 'C' for 6.002x) or None
+    **Example GET Response**
+        [{
+            "username": "bob",
+            "email": "bob@example.com",
+            "course_id": "course-v1:edX+DemoX+Demo_Course",
+            "passed": false,
+            "percent": 0.03,
+            "letter_grade": null,
+        },
+        {
+            "username": "fred",
+            "email": "fred@example.com",
+            "course_id": "course-v1:edX+DemoX+Demo_Course",
+            "passed": true,
+            "percent": 0.83,
+            "letter_grade": "B",
+        },
+        {
+            "username": "kate",
+            "email": "kate@example.com",
+            "course_id": "course-v1:edX+DemoX+Demo_Course",
+            "passed": false,
+            "percent": 0.19,
+            "letter_grade": null,
+        }]
+    """
+    permission_classes = (IsUserInUrlOrStaff,)
+
+    def get(self, request, course_id=None):
+        """
+        Gets a course progress status.
+        Args:
+            request (Request): Django request object.
+            course_id (string): URI element specifying the course location.
+        Return:
+            A JSON serialized representation of the requesting user's current grade status.
+        """
+        username = request.GET.get('username')
+
+        if not course_id:
+            course_id = request.GET.get('course_id')
+
+        course_key = self._get_course_key(course_id)
+        if isinstance(course_key, Response):
+            # Returns a 404 if course_id is invalid, or request.user is not enrolled in the course
+            return course_key
+
+        if username:
+            grade_user = self._get_effective_user(request, course_key)
+            if isinstance(grade_user, Response):
+                # Returns a 404 if the requested user does not exist or is not enrolled
+                return grade_user
+
+            course_grade = CourseGradeFactory().read(grade_user, course_key=course_key)
+
+            return Response([
+                self._make_grade_response(grade_user, course_key, course_grade)
+            ])
+        else:
```

Comment:
Nit: For improved code readability, it would be ideal if the code in the `if` clause and the code in the `else` clause were in their own smaller helper methods.  That would keep this main method smaller - and easier to read.